internal/tests/api/wallet_test.go

```diff
@@ -116,6 +174,126 @@ func TestWalletBalance(t *testing.T) {
 	t.Logf("Received correct balance (%s)", response.Data.Balance.String())
 }
 
+func TestWalletSignMessage(t *testing.T) {
+	// Take a snapshot, revert at the end
+	snapshotName, err := testMgr.CreateCustomSnapshot(osha.Service_EthClients | osha.Service_Filesystem)
+	if err != nil {
+		fail("Error creating custom snapshot: %v", err)
+	}
+	defer wallet_cleanup(snapshotName)
+
+	// Commit a block just so the latest block is fresh - otherwise the sync progress check will
+	// error out because the block is too old and it thinks the client just can't find any peers
+	err = testMgr.CommitBlock()
+	if err != nil {
+		t.Fatalf("Error committing block: %v", err)
+	}
+
+	// Regen the wallet
+	apiClient := testMgr.GetApiClient()
+	derivationPath := string(wallet.DerivationPath_Default)
+	index := uint64(0)
+	_, err = apiClient.Wallet.Recover(&derivationPath, keys.DefaultMnemonic, &index, goodPassword, true)
+	require.NoError(t, err)
+	t.Log("Recover called")
+
+	message := []byte("hello world")
+	response, err := apiClient.Wallet.SignMessage(message)
+	require.NoError(t, err)
+	t.Log("SignMessage called")
+
+	require.NotEmpty(t, response.Data.SignedMessage)
+	prefix := fmt.Sprintf("\x19Ethereum Signed Message:\n%d", len(message))
```

Comment:
You don't have to do this manually - the Geth library can do it for you. See:

- https://github.com/nodeset-org/nodeset-svc-mock/blob/4f7aeb08967ec1234ba510db24ec26977851a6ee/auth/auth.go#L117
- https://github.com/nodeset-org/nodeset-svc-mock/blob/4f7aeb08967ec1234ba510db24ec26977851a6ee/auth/auth.go#L100

Reply:
fixed

Reply:
I didnt use CreateSignature from svc-mock here since i wanted the backend to generate the signature (it's what i'm testing)